process/smartContract/process.go

```diff
@@ -750,6 +751,27 @@ func (sc *scProcessor) processVMOutput(
 		return nil, nil, process.ErrNilTransaction
 	}
 
+	if isTooMuchGasProvided(vmOutput.GasRemaining, gasProvided) {
```

Comment:
Maybe reuse an integration test for this scenario?

Reply:
will do after system tests will pass

Reply:
Ok